openmdao/core/tests/test_group.py

```diff
@@ -2651,8 +2651,9 @@ def test_override_units(self):
         g1.set_input_defaults('x', units='inch', val=2.)
 
         msg = "Groups 'G1' and 'G1.G2' called set_input_defaults for the input 'x' with conflicting 'units'. The value (inch) from 'G1' will be used."
+        p.setup(check=True)
         with assert_warning(UserWarning, msg):
```

Comment:
After changing the check to use the logger (see other comment), you'll have to change this test.  To see other examples, search the source for 'TestLogger'.